content/blog/2023/07/22/2023-07-22-gsoc-2023-midterm.adoc

```diff
@@ -0,0 +1,188 @@
+---
+layout: post
+title: Jenkins in Google Summer of Code Midterm Recap
+tags:
+- gsoc
+- gsoc2023
+- midterm
+authors:
+- ash-sxn
+- harsh-ps-2003
+- jagruti
+- vandit1604
+opengraph:
+  image: /images/gsoc/opengraph.png
+sig:
+- gsoc
+---
+
+The Jenkins in GSoC program recently reached its midterm milestone for GSoC 2023.
+This milestone served to assess performance, and precipitate pass/fail decisions.
+Taking time to evaluate the progress and workflow of the project provides an opportunity to correct course and address underlying issues.
+
+Earlier this month, on the link:https://www.meetup.com/jenkins-online-meetup/[Jenkins Online Meetup], GSoC contributors presented midterm status and progress for their projects, below is a recap.
+Thank you Jagruti, Harsh, Vandit, and Ashutosh for your contributions to Jenkins!
+
+== link:https://www.jenkins.io/projects/gsoc/2023/projects/add-probes-to-plugin-health-score/[Add Probes to "Plugin Health Score"]
+By link:https://github.com/Jagrutiti/[Jagruti Tiwari]
+
+A probe collects data about each plugin.
+It is an automated system that ensures the Jenkins Community-defined rules are followed.
+The probe data is used by health scoring to compute a score, and a higher-quality probe receives a good score.
+Conversely, an obsolete or outdated plugin is given a lower score.
+
+Sharing details about the probes that I worked on during the first phase of GSoC.
+
+=== Unreleased Production Changes Probe
+
+A bug fix, security improvement, or new feature can only be beneficial when it's actually released.
+The "Unreleased Production Changes" probe is designed to detect unreleased commits in the production files.
+
+This probe's primary goal is to identify unreleased production commits specifically in the Maven production files.
+
+Following my mentors' advice, I began by writing test cases to accurately reproduce the probe's requirements. Subsequently, I added the necessary code to pass each of these test cases.
+
+During the implementation, I utilized the JGit library and familiarized myself with its features.
+However, I encountered some challenges due to a lack of depth in Git concepts, which required me to delve deeper and strengthen my understanding of Git concepts along the way.
```

Comment:
Read the link:blog/2023/07/16/unreleased-production-changes-probe.adoc[unreleased production changes] blog post for more details on the probe.

Reply:
Updated